source/CommonJobs/CommonJobs.Application.EvalForm/Commands/GetEvaluationCalifications.cs

```diff
@@ -49,42 +49,113 @@ public override CalificationsDto ExecuteWithResult()
                 .Customize(x => x.WaitForNonStaleResultsAsOfLastWrite())
                 .Where(x => x.EvaluationId == evId).ToList();
 
-            if (evaluation.ReadyForDevolution) // Ready for devolution
+            if (!CanViewEvaluation(evaluationDto, califications))
             {
-                return new CalificationsDto()
-                {
-                    View = UserView.Company,
-                    Evaluation = evaluationDto,
-                    Califications = califications.Where(c => (_loggedUser == c.EvaluatorEmployee && c.EvaluatorEmployee == c.EvaluatedEmployee) || c.Owner == CalificationType.Company).ToList()
-                };
+                throw new ApplicationException(string.Format("Error: Usuario {0} no tiene permiso para ver la evaluación {1}", _loggedUser, evId));
             }
-            else if (_evaluatedUser == _loggedUser) // Auto-evaluation
+
+            // If evaluation is finished
+            // - Show ALWAYS company & auto
+
+            if (evaluation.Finished)
             {
-                return new CalificationsDto()
-                {
-                    View = UserView.Auto,
-                    Evaluation = evaluationDto,
-                    Califications = new List<EvaluationCalification>() { califications.Single(c => _loggedUser == c.EvaluatorEmployee && c.EvaluatorEmployee == c.EvaluatedEmployee) }
-                };
+                return GetFinishedEvaluation(evaluationDto, califications);
             }
-            else if (evaluation.ResponsibleId == _loggedUser) // Responsible & Company (responsible && finished)
+
+            // If loggedUser is the evaluated
+            // - Show auto-eval if not RFD
+            // - Show auto-eval & company if RFD
+
+            if (_loggedUser == _evaluatedUser)
+            {
+                return GetAutoEvaluation(evaluationDto, califications);
+            }
+
+            // If loggedUser is responsible
+            // - Show all
+
+            if (_loggedUser == evaluationDto.ResponsibleId)
+            {
+                return GetEvaluation(evaluationDto, califications);
```

Comment:
What do you think about a more representative name like `GetFullEvaluation` or `GetResponsibleEvaluation`?